builder/lifecycle.go

```diff
@@ -16,44 +16,42 @@ import (
 )
 
 var (
-	apiVersionAssumed = api.MustParse("0.1")
-	apiVersionLatest  = api.MustParse("0.2")
-
-	v0_3_0 = semver.MustParse("0.3.0")
-	v0_4_0 = semver.MustParse("0.4.0")
-
-	lifecycleVersionAssumed = &Version{Version: *v0_3_0}
-	lifecycleVersionLatest  = &Version{Version: *v0_4_0}
+	v0_3_0 = *semver.MustParse("0.3.0")
+	v0_4_0 = *semver.MustParse("0.4.0")
+
+	lifecycleBinaries = []string{
+		"detector",
+		"restorer",
+		"analyzer",
+		"builder",
+		"exporter",
+		"cacher",
+		"launcher",
+	}
 )
 
-var AssumedLifecycleDescriptor = LifecycleDescriptor{
-	Info: LifecycleInfo{
-		Version: lifecycleVersionAssumed,
-	},
-	API: LifecycleAPI{
-		PlatformVersion:  apiVersionAssumed,
-		BuildpackVersion: apiVersionAssumed,
-	},
-}
-
-var LatestLifecycleDescriptor = LifecycleDescriptor{
-	Info: LifecycleInfo{
-		Version: lifecycleVersionLatest,
-	},
-	API: LifecycleAPI{
-		PlatformVersion:  apiVersionLatest,
-		BuildpackVersion: apiVersionLatest,
-	},
+func AssumedLifecycleDescriptor() LifecycleDescriptor {
```

Comment:
> I'm not sure what the motivation for moving this to a function is?

The motivation for the function is to handle pointer references in cases where we use the value as a "default" before deserialization. Through the process of "decoding" it would update the pointer reference instead of creating a new reference hence affecting multiple objects. Basically, we decided to make this a function to ensure we always have a new reference. This would hopefully prevent us from running into this issue again.

That said, we could probably just go with the separate constants as strings to prevent the same issue.